pyne/alara.py

```diff
@@ -94,17 +96,28 @@ def mesh_to_fluxin(flux_mesh, flux_tag, fluxin="fluxin.out",
     # write flux data block by block
     with open(fluxin, "w") as f:
         if not sub_voxel:
-            with Bar("Writing alara fluxin", max=len(flux_mesh), suffix='%(percent).1f%% - %(eta)ds') as bar:
+            if print_progress:
+                bar = Bar("Writing alara fluxin", max=len(flux_mesh), suffix='%(percent).1f%% - %(eta)ds')
                 for i, mat, ve in flux_mesh:
                     f.write(_output_flux_block(ve, tag_flux, reverse))
                     bar.next()
+                bar.finish()
+            else:
+                for i, mat, ve in flux_mesh:
+                    f.write(_output_flux_block(ve, tag_flux, reverse))
```

Comment:
I see that's a little messy!

This might work to clean it up a little bit
```suggestion
            bar = Bar("Writing alara fluxin", max=len(flux_mesh), suffix='%(percent).1f%% - %(eta)ds') if print_progress else None
            for i, mat, ve in flux_mesh:
               f.write(_output_flux_block(ve, tag_flux, reverse))
               bar.next() if print_progress else None
            bar.finish() if print_progress else None
```

Another option would be to extend Bar?

```
class IfBar(....., show_bar = True):

    def init(....):

        self.show_bar = show_bar

        if not self.show_bar:
            pass

        super().__init__(....)

    def next():
        if self.show_bar:
            super().next()

    def finish():
        if self.show_bar:
            super().finish()
```

Reply:
Agreed. The turn on/off of the print progress requires more indents. The `IfBar` is add in #1409 .